memo_app.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'cgi'
+require 'json'
+require 'securerandom'
+
+MEMOS_FILE = 'memos.json'
+
+not_found do
+  'ページが見つかりません!'
+end
+
+def load_memos
+  JSON.parse(File.read(MEMOS_FILE), symbolize_names: true)
+end
+
+def add_memos(params)
+  existing_data = load_memos
+  existing_data << { id: SecureRandom.uuid, title: params[:title], content: params[:content], created_at: Time.now, updated_at: Time.now }
```

Comment:
```suggestion
  existing_data[SecureRandom.uuid] = { title: params[:title], content: params[:content], created_at: Time.now, updated_at: Time.now }
```

現状はデータが配列で格納されていますが、uuidをキーにしたハッシュにした方が他の処理もスマートに書けそうです。

Reply:
uuidをキーにしたハッシュにすることで、find_memoメソッド内の処理において、```find```を使用せずに、```params[:id]```のみを用いて該当データを参照できるようになりました。